pkg/operator/ceph/object/admin.go

```diff
@@ -149,3 +167,85 @@ func isInvalidFlagError(err error) bool {
 	// it's also returned from some other failures, but this should be rare for Rook
 	return exitCode == 22
 }
+
+func (r *ReconcileCephObjectStore) reconcileRGWAdminOpsUser(cephObjectStore *cephv1.CephObjectStore) error {
+	// Fetch the secret for admin ops user
+	s := &v1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: rgwAdminOpsUserSecretName, Namespace: cephObjectStore.Namespace}, s)
+	if err != nil && kerrors.IsNotFound(err) {
+		if cephObjectStore.Spec.IsExternal() {
+			return errors.Errorf("failed to find rgw admin ops secret %q", rgwAdminOpsUserSecretName)
+		}
+		// Go and create the secret
+		err = r.createRGWAdminOpsUser(cephObjectStore)
+		if err != nil {
+			return errors.Wrap(err, "failed to create rgw admin ops user")
+		}
+	}
+	if _, ok := s.Data[rgwAdminOpsUserAccessKey]; !ok {
+		return errors.Errorf("failed to find accessKey for rgw admin ops in secret %q", rgwAdminOpsUserSecretName)
+	}
+	if _, ok := s.Data[rgwAdminOpsUserSecretKey]; !ok {
+		return errors.Errorf("failed to find secretKey for rgw admin ops in secret %q", rgwAdminOpsUserSecretName)
+	}
+
+	// Set the keys for further usage
+	r.rgwAdminOpsUserAccessKey = string(s.Data[rgwAdminOpsUserAccessKey])
+	r.rgwAdminOpsUserSecretKey = string(s.Data[rgwAdminOpsUserSecretKey])
+
+	return nil
+}
+
+func (r *ReconcileCephObjectStore) createRGWAdminOpsUser(cephObjectStore *cephv1.CephObjectStore) error {
+	objContext := NewContext(r.context, r.clusterInfo, cephObjectStore.Name)
+	userConfig := ObjectUser{
+		UserID:      rgwAdminOpsUserSecretName,
+		DisplayName: &rgwAdminOpsUserDisplayName,
+	}
+	logger.Debugf("creating s3 user object %q for object store %q", userConfig.UserID, cephObjectStore.Namespace)
+	user, rgwerr, err := CreateUser(objContext, userConfig)
+	if err != nil {
+		if rgwerr == ErrorCodeFileExists {
+			user, _, err = GetUser(objContext, userConfig.UserID)
```

Comment:
Instead of storing the admin ops user creds in a secret, can we just query them from Ceph when needed if it's a local ceph cluster? We are anyways querying them from ceph here, right?

Reply:
Yes we are. Sounds good to me.